src/core/reducers/user.js

```diff
@@ -45,6 +45,21 @@ export const selectDisplayName = (state: { user: UserStateType }) => {
   return state.user.username;
 };
 
+export const hasPermission = (
+  state: { user: UserStateType }, permission: string,
+): boolean => {
+  const permissions = state.user.permissions;
+  if (!permissions) {
+    return false;
+  }
+
+  return permissions.includes(permission);
```

Comment:
It's going to be easier to scale this up if you include the admin check as an implementation detail:

```js
return permissions.includes(permission) || isAdmin(state);
```

Admins can do anything.

Reply:
ah that makes sense indeed!

Reply:
I removed the `isAdmin` for now as we're not using it.